linkml/utils/pattern.py

```diff
@@ -0,0 +1,34 @@
+from linkml_runtime.utils.schemaview import SchemaView
+
+
+def materialize_patterns(schema_view: SchemaView):
+    """Materializes slot patterns based on structured
+    patterns in the settings.
+
+    Note: Changes the schema in place.
+
+    :param schema_view: SchemaView object with LinkML YAML
+        already loaded
+    """
+
+    # fetch settings from schema_view
+    settings_dict = schema_view.schema.settings
+
+    # dictionary of key and string value of settings dict
+    format_spec = {}
+
+    for k, setting in settings_dict.items():
+
+        # create spec dictionary with keys that will replace
+        # substrings in the structured pattern syntax
+        format_spec["{" + k + "}"] = setting.setting_value
+
+    for _, slot_defn in schema_view.all_slots().items():
+        if slot_defn.structured_pattern:
+            struct_pat = slot_defn.structured_pattern
+
+            # compute pattern from structured patterns
+            # and format_spec dictionary
+            slot_defn.pattern = " ".join(
+                format_spec.get(pat, pat) for pat in struct_pat.syntax.split()
```

Comment:
I'm not sure how split() is working here, but has there been discussion about what should be allowed in struct_pat.syntax?  I think we should allow for "{number}{unit}" i.e. allow for no spaces - many systems might have this content. Means also that slot_defn.pattern parsing will have to preserve spaces existing between structured_pattern "{a} {b} {c}{d}{e}" etc. values.

Reply:
A quickie way to do this is (for interpolated strings only):

slot_defn.pattern = struct_pat.syntax.format(**settings_dict)

so you can probably avoid both loops above.

Reply:
@ddooley: Thank you for the review.

@cmungall had the same comment as you, i.e., what if the serialization isn't space separated. We need to have provisions for handling `{foo}{bar}`.

However the problem with implementing it as:

```
slot_defn.pattern = struct_pat.syntax.format(**format_spec)
```

is that the dictionary keys have a `.` dot in them, and there's no way to escape dots apparently: https://stackoverflow.com/questions/23909449/how-to-escape-dot-in-python-str-format